src/resolve-dependency.js

```diff
@@ -78,8 +78,8 @@ function getExportsTarget (exports, conditions, cjsResolve) {
   }
   else if (Array.isArray(exports)) {
     for (const item of exports) {
-      const target = getExportsTarget(item);
-      if (target !== undefined)
+      const target = getExportsTarget(item, conditions, cjsResolve);
+      if (target !== undefined && target !== null && target.startsWith('./'))
```

Comment:
```suggestion
      if (typeof target === 'string' && target.startsWith('./'))
```

Reply:
Ah, yes this is better :)

Reply:
Actually sorry you're completely right - it's actually incorrect that this is different to the below, pushing up a fix.